### PR TITLE
Add `exclude_status` param and tests

### PR DESCRIPTION
**Pull request by albarin**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR introduces the new `exclude_status` parameter to the `/products` endpoint.
It allows filtering out products by multiple statuses simultaneously: `draft`, `pending`, `private`, and `publish`.

With this implementation, `exclude_status` takes precedence over `status` and `include_status` if any of them is present in the same request.

Part of #52773

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure you have multiple products in different statuses (`draft`, `pending`, `private`, and `publish`).
2. Go to `wp-admin/admin.php?page=wc-settings&tab=advanced&section=keys` and add a new key with `Read/Write` permissions.
3. Make the following request using the consumer key and secret generated in the previous step, adding the `include_status` param:
```bash
curl --insecure -u [KEY]:[SECRET] --request GET --url 'https://[STORE_URL]/wp-json/wc/v3/products?exclude_status=draft'
```
4. Ensure the result returns only products with a status different than `draft`.
5. Repeat the request with multiple statuses (comma-separated) and check the response makes sense.
6. Try with a non-existing status, it should return a 400 response.
7. Try any other combinations that you can think of, and check the response is consistent.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Add new "exclude_status" filter on the "/products" endpoint
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php [hunk 1]
```diff
@@ -178,6 +185,10 @@ protected function prepare_objects_query( $request ) {
 			$args['post_status'] = $request['include_status'];
 		}
 
+		if ( ! empty( $request['exclude_status'] ) ) {
+			$this->exclude_status = $request['exclude_status'];
+		}
```
Comment (coreymckrill) [inline plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php]: 🤔 I wonder if there could be a scenario where the `exclude_status` prop gets populated here by a request object, but then `get_objects` is never called so the prop never gets cleared, and then a second request object gets processed and inherits this prop from the first request object... I guess that is a pretty extreme edge case. But one way to mitigate that risk wouuld be to clear the prop here when `$request['exclude_status']` _is_ empty.

**Review by coreymckrill — APPROVED** [review]
👍 Works as described, very thorough unit tests, nice work.

Testing notes:
* Tried with a single status, and with multiple statuses
* Tried with the trash status
* Tried with `any` (invalid as expected)

Approving but not merging in case you want to discuss one minor, optional comment below.

**Review by AnirudhBhat — COMMENTED** [review]
Thanks for the changes @albarin! I tested the changes and it works as expected :shipit:

**Comment by iamgabrielma** [comment]
Tested the branch as well and worked nicely 🎉 

✅  single exclusion
✅  multiple exclusion
✅  non-existent status returns 400 with `exclude_status[0] is not one of trash, draft, pending, private, and publish.`

File: plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php (hunk 1 above)
Comment (albarin) [reply plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php]: Good catch! Updated here 31167d40417d11fedbef48dbdb2b197e7bb0bbc0

**Pull request merged**